src/sidebar/components/test/markdown-editor-test.js

```diff
@@ -123,7 +123,7 @@ describe('MarkdownEditor', () => {
       });
 
       if (key) {
-        it('applies formatting when shortcut key is pressed', () => {
+        it('applies formatting when CTRL+shortcut key is pressed', () => {
```

Comment:
I would have preferred not to duplicate the entire test function just to change one object property in this case as it makes it less clear that the behavior is supposed to be exactly the same in both cases. Maybe extract a `testShortcutKey(keyEventArgs)` helper function?

Reply:
We discussed this in person on a call: refactoring these tests to DRY them out is unexpectedly fiddly. Though there's lots of repeated code, the state that informs the different repeated code bits is different between the tests enough that extracting into a shared function or two proved convoluted and resulted in something hard to read and make sense of.

Reply:
Hi Lyza,

I don't think the DRY needs to be elaborate here. This sketch shows one possible way of doing what I had in mind:

```diff
diff --git a/src/sidebar/components/test/markdown-editor-test.js b/src/sidebar/components/test/markdown-editor-test.js
index 48bb6adbe..e10f6e02d 100644
--- a/src/sidebar/components/test/markdown-editor-test.js
+++ b/src/sidebar/components/test/markdown-editor-test.js
@@ -123,52 +123,35 @@ describe('MarkdownEditor', () => {
       });
 
       if (key) {
-        it('applies formatting when CTRL+shortcut key is pressed', () => {
-          const onEditText = sinon.stub();
-          const text = 'toolbar shortcut test';
-          const wrapper = createComponent({ text, onEditText });
-          const input = wrapper.find('textarea');
-          input.getDOMNode().selectionStart = 0;
-          input.getDOMNode().selectionEnd = text.length;
-
-          input.simulate('keydown', {
-            ctrlKey: true,
-            key,
+        const shortcutKeyEventDetails = [{
+          ctrlKey: true,
+          key,
+        },{
+          metaKey: true,
+          key,
+        }];
+
+        shortcutKeyEventDetails.forEach(eventDetails => {
+          it('applies formatting when shortcut key is pressed', () => {
+            const onEditText = sinon.stub();
+            const text = 'toolbar shortcut test';
+            const wrapper = createComponent({ text, onEditText });
+            const input = wrapper.find('textarea');
+            input.getDOMNode().selectionStart = 0;
+            input.getDOMNode().selectionEnd = text.length;
+
+            input.simulate('keydown', eventDetails);
+
+            assert.calledWith(onEditText, {
+              text: 'formatted text',
+            });
+            const [formatFunction, ...args] = effect;
+            assert.calledWith(
+              formatFunction,
+              sinon.match({ text, selectionStart: 0, selectionEnd: text.length }),
+              ...args
+            );
           });
-
-          assert.calledWith(onEditText, {
-            text: 'formatted text',
-          });
-          const [formatFunction, ...args] = effect;
-          assert.calledWith(
-            formatFunction,
-            sinon.match({ text, selectionStart: 0, selectionEnd: text.length }),
-            ...args
-          );
-        });
-
-        it('applies formatting when META+shortcut key is pressed', () => {
-          const onEditText = sinon.stub();
-          const text = 'toolbar shortcut test';
-          const wrapper = createComponent({ text, onEditText });
-          const input = wrapper.find('textarea');
-          input.getDOMNode().selectionStart = 0;
-          input.getDOMNode().selectionEnd = text.length;
-
-          input.simulate('keydown', {
-            metaKey: true,
-            key,
-          });
-
-          assert.calledWith(onEditText, {
-            text: 'formatted text',
-          });
-          const [formatFunction, ...args] = effect;
-          assert.calledWith(
-            formatFunction,
-            sinon.match({ text, selectionStart: 0, selectionEnd: text.length }),
-            ...args
-          );
         });
       }
     });
```